app/agreement/submit.js

```diff
@@ -2,15 +2,18 @@ const cache = require('../cache')
 const save = require('./save')
 const { post } = require('../api/agreement')
 const util = require('util')
+const status = require('./status')
 
 const submit = async (request) => {
   await save(request)
   const { agreement } = await cache.get(request)
   const { agreementNumber, organisation } = agreement
 
+  const submitted = status.find(x => x.name === 'Submitted')
+
   const url = '/agreement/submit'
   await post(url, { agreementNumber, sbi: organisation.sbi })
-  await cache.update(request, { agreement: { submitted: true } })
+  await cache.update(request, { agreement: { submitted: true, statusId: submitted } })
```

Comment:
Think this needs to be `submitted.id` as done in `agreement/create.js`

Reply:
great spot, thanks!